addons/website/models/ir_ui_view.py

```diff
@@ -25,8 +25,19 @@ class View(models.Model):
     first_page_id = fields.Many2one('website.page', string='Website Page', help='First page linked to this view', compute='_compute_first_page_id')
     track = fields.Boolean(string='Track', default=False, help="Allow to specify for one page of the website to be trackable or not")
     visibility = fields.Selection([('', 'All'), ('connected', 'Signed In'), ('restricted_group', 'Restricted Group'), ('password', 'With Password')], default='')
-    visibility_group = fields.Many2one('res.groups')
-    visibility_password = fields.Char(groups='website.group_website_publisher')
+    visibility_group = fields.Many2one('res.groups', copy=False)
+    visibility_password = fields.Char(groups='base.group_system', copy=False)
+    visibility_password_display = fields.Char(compute='_get_pwd', inverse='_set_pwd', groups='website.group_website_designer')
+
+    @api.depends('visibility_password')
+    def _get_pwd(self):
+        for r in self:
+            r.visibility_password_display = r.sudo().visibility_password and '********' or ''
+
+    def _set_pwd(self):
+        for r in self:
+            r.with_user(SUPERUSER_ID).visibility_password = tools.misc._hash(r.env, r.visibility_password_display, r.id, r.id)
```

Comment:
Wouldn't using a KDF (as with user passwords) be more appropriate than using a MAC? Passlib is a hard dependency so there's no problem using it (though there's [a small annoyance when initially setting the password](https://github.com/odoo/odoo/blob/a0f0cdd70e88e9da322b9e02157c58e233ba7104/odoo/addons/base/models/res_users.py#L296) because of deprecations).

Could use the same context as user passwords, or could use a more relaxed number of rounds [as we're doing for the API keys](https://github.com/odoo/odoo/commit/1f5f76120736edbd3ac4e73a32955e85095e6aa8#diff-c087b09b8455bbf3fa546a56c6e271bcR1421-R1426).